state-chain/pallets/cf-elections/src/electoral_systems/blockchain/delta_based_ingress.rs

```diff
@@ -174,116 +174,84 @@ where
 			};
 
+			let mut new_totals = Vec::new();
 			let mut closed_channels = Vec::new();
-			for (account, (details, _)) in &channels {
-				let (
-					option_ingress_total_before_chain_tracking,
-					option_ingress_total_after_chain_tracking,
-				) = match option_consensus.as_ref().and_then(|consensus| consensus.get(account)) {
-					None => (None, None),
-					Some(consensus_ingress_total) => {
-						if consensus_ingress_total.block_number <= *chain_tracking {
-							(Some(*consensus_ingress_total), None)
-						} else {
-							match pending_ingress_totals.remove(account) {
-								None => (None, Some(*consensus_ingress_total)),
-								Some(pending_ingress_total) => {
-									if pending_ingress_total.block_number <
-										consensus_ingress_total.block_number &&
-										pending_ingress_total.amount <
-											consensus_ingress_total.amount
-									{
-										if pending_ingress_total.block_number <= *chain_tracking {
-											(
-												Some(pending_ingress_total),
-												Some(*consensus_ingress_total),
-											)
-										} else {
-											(None, Some(pending_ingress_total))
-										}
-									} else {
-										(None, Some(*consensus_ingress_total))
-									}
-								},
-							}
-						}
+
+			for (
+				account,
+				(
+					OpenChannelDetails { asset, close_block },
+					ChannelTotalIngressed { amount: initial_amount, .. },
+				),
+			) in &properties
+			{
+				if let Some(
+					new_total @ ChannelTotalIngressed {
+						block_number: ingress_block,
+						amount: ingress_amount,
 					},
-				};
-
-				if let Some(ingress_total) = option_ingress_total_before_chain_tracking {
-					let previous_amount = electoral_access
-						.unsynchronised_state_map(&(account.clone(), details.asset))?
-						.map_or(Zero::zero(), |previous_total_ingressed| {
-							previous_total_ingressed.amount
-						});
-					match previous_amount.cmp(&ingress_total.amount) {
+				) = option_consensus
+					.as_ref()
+					.and_then(|consensus| consensus.get(account))
+					.filter(|totals| totals.block_number <= *chain_tracking)
+				{
+					match initial_amount.cmp(&ingress_amount) {
 						Ordering::Less => {
 							Sink::on_ingress(
 								account.clone(),
-								details.asset,
-								ingress_total.amount - previous_amount,
-								ingress_total.block_number,
+								*asset,
+								*ingress_amount - *initial_amount,
+								*ingress_block,
 								(),
 							);
 							electoral_access.set_unsynchronised_state_map(
-								(account.clone(), details.asset),
-								Some(ingress_total),
+								(account.clone(), *asset),
+								Some(*new_total),
 							)?;
+							new_totals.push((account.clone(), *new_total));
 						},
 						Ordering::Greater => {
 							Sink::on_ingress_reverted(
 								account.clone(),
-								details.asset,
-								ingress_total.amount - previous_amount,
+								*asset,
+								*ingress_amount - *initial_amount,
 							);
```

Comment:
```suggestion
							new_totals.push((account.clone(), *new_total));
```
I think we also want to update the election properties when the amount decreased, since this is the value we had consensus on.

Reply:
I'm actually not all sure if/how we can come to consensus on a decrease...